### PR TITLE
Added the `uniffi_reexport_scaffolding` macro

### PR DESCRIPTION
**Pull request by bendk**
This can be used to work around [rust-lang#50007](https://github.com/rust-lang/rust/issues/50007), which is getting to be more and more of an issue on the desktop JS project.

### CODE REVIEW

File: uniffi_bindgen/src/scaffolding/templates/TopLevelFunctionTemplate.rs [hunk 1]
```diff
@@ -14,3 +15,24 @@ pub extern "C" fn {{ func.ffi_func().name() }}(
...
+
+// Create a macro that can be used by a dependent create to forward all of the scaffolding
+// functions defined here.  This is used in 2 ways:
+//
+//   - Building libraries that contain multiple UniFFI components (megazord, libxul).  The combined
+//     library can depend on each component then call uniffi_forward_scaffolding!() for each one.
+//   - Test fixtures.  Each binding generator needs to test against the test fixtures.  To allow
+//     this, they call uniffi_forward_scaffolding!() for the test fixture in their test module.
+#[macro_export]
+macro_rules! uniffi_forward_scaffolding {
```
Comment (tarikeshaq) [inline uniffi_bindgen/src/scaffolding/templates/TopLevelFunctionTemplate.rs:28]: Hmm I'm thinking about non-high level function forwarding (like forwarding to a method on an object) - how would that work? or would forwarding just the high-level functions be enough for our fixtures?

**Review by tarikeshaq — COMMENTED** [review]
Thanks Ben!! Had a question about methods, but otherwise, this looks good!  can we add a test to validate that the macro works properly?

We could have one of our fixtures depend on another or maybe make a new crate that depends on one of the existing ones

File: uniffi_bindgen/src/scaffolding/templates/TopLevelFunctionTemplate.rs (hunk 1 above)
Comment (bendk) [reply uniffi_bindgen/src/scaffolding/templates/TopLevelFunctionTemplate.rs]: Great point.  I thought I was getting those too, but I wasn't.  Just pushed a new commit that I think should do this properly.

File: uniffi_bindgen/src/scaffolding/templates/ForwardUniFFIScaffolding.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Create a macro that can be used by a dependent create to forward all of the scaffolding
```
Comment (mhammond) [inline uniffi_bindgen/src/scaffolding/templates/ForwardUniFFIScaffolding.rs:1]: We might as well be extra redundant and link to the rust-lang issue here too, so the reason for this file isn't a mystery unless you have the additional context.

**Review by mhammond — COMMENTED** [review]
This LGTM, but I agree with Tarik that a test for all the cases this covers would be ideal.

**Comment by bendk** [comment]
I tried adding unit tests and got a stack overflow when calling the scaffolding functions.  I'm pretty sure I manually tested it, but this whole issue is very strange.

Instead of trying to define new functions, what if we make with the current hack the official way to do this?  I added a helper macro to assist and some unit tests that hopefully will break if the hack stops working.
